src/IO/ADIOS/ADIOS2IOHandler.cpp

```diff
@@ -990,11 +1098,12 @@ namespace detail
             throw std::runtime_error(
                 "[ADIOS2] Internal error: Could not create Variable '" + name + "'." );
         }
-        // check whether the unique_ptr has an element
-        // and whether the held operator is valid
-        if ( compression && *compression )
+        for( auto const & compression : compressions )
         {
-            var.AddOperation( *compression );
+            if( compression.op )
+            {
+                var.AddOperation( compression.op, compression.params );
```

Comment:
Just to make sure this works:
How would I set parameters of blosc, e.g.
```
    // add operations
    adios2::Operator BloscOp =
        adios.DefineOperator("BloscCompressor", adios2::ops::LosslessBlosc);

    varBig.AddOperation(BloscOp,
                         {{adios2::ops::blosc::key::compressor, "zstd"}});
    varBig.AddOperation(BloscOp,
                         {{adios2::ops::blosc::key::clevel, "1"}});
    varBig.AddOperation(BloscOp,
                         {{adios2::ops::blosc::key::doshuffle, "BLOSC_BITSHUFFLE"}});
```

via json?

Reply:
I updated [the test](https://github.com/openPMD/openPMD-api/blob/f21074558c308398e6dda2cca3507a423db95ee0/test/SerialIOTest.cpp#L2166) accordingly. I skipped setting the compressor to `zstd` since ADIOS2 will fail if that is not available.

Reply:
Okay... but it's our default compressor we use all the time in production. Maybe add it in some way...

@chuckatkins @guj is it possible to query the ADIOS2 API at runtime to be sure that e.g. blosc was compiled in?

Reply:
Hi Axel,  Chuck will add it in.

Reply:
@guj do we already have an issue or PR to link here, for reference?